File: Java/백준/Silver/2108. 통계학/통계학.java
import java.io.*;
import java.util.Arrays;

public class Main {
    // 2108 통계학
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int N = Integer.parseInt(bf.readLine());

        //입력
        int arr[] = new int[N];
        arr[0] = Integer.parseInt(bf.readLine());
        int max = arr[0], min = arr[0];

        for (int i = 1; i < N; i++) {
            arr[i] = Integer.parseInt(bf.readLine());
            if (arr[i] >= max) max = arr[i]; //최대값 구하기
            else if (arr[i] <= min) min = arr[i]; //최소값 구하기
        }
        //-----

        int hap = 0;
        int count[] = new int[max - min + 1 ];
        for (int v : arr) {
            count[v - min]++; // 원소의 개수 카운트
            hap += v; // 원소들의 합을 구함
        }

        //
        int count_max = 0;
        for (int i = 0; i < count.length; i++) {
            if (count[i] >= count_max) {
                count_max = count[i];
            }
        }

        int mode[] = new int[N]; // 최빈값 리스트
        Arrays.fill(mode, 4001);
        int index = 0;
        for (int i = 0; i < count.length; i++) {
            if (count[i] == count_max) { //<- 이부분 고쳐야함. 4월 15일에 때려침....
                mode[index] = i + min;
                index++;
            }
        }
        Arrays.sort(mode);
        //System.out.println(count_max);
        //System.out.println(Arrays.toString(mode));
        int sorted[] = new int[N];
        for (int i = 1; i < count.length; i++) {
            count[i] += count[i - 1];
        }

        for (int v : arr) {
            sorted[count[v - min] - 1] = v;
            count[v - min]--;
        }
        bw.write(Math.round((float) hap / arr.length) + "\n"); //산술평균
        bw.write(sorted[sorted.length / 2] + "\n"); // 중앙값
        bw.write((index == 1 ? mode[0] : mode[1]) + "\n"); // 최빈값
        bw.write(max + (-min) + "\n"); // 범위

        bw.flush();
    }
}

File: Python/백준/Silver/2108. 통계학/통계학.py
import sys

n = int(sys.stdin.readline())
arr = [int(sys.stdin.readline()) for _ in range(n)] # 수들이 담길 배열
arr = sorted(arr)
count_arr = [0 for _ in range(abs(arr[0]-arr[n-1])+1)] # 최솟값부터 최댓값까지 개수를 셀 배열(0포함)

for num in arr:
    count_arr[num+arr[0]*(-1)] += 1 # -1을 곱하여 부호를 반전시킴 -2가 최솟값이라면 index 0이 -2이기 때문

max_count = max(count_arr)
max_nums = []
for i in range(len(count_arr)):
    if count_arr[i] == max_count:
        max_nums.append(i+arr[0]) # 원래 값으로 바꿈

print(round(sum(arr)/n))    # 산술평균
print(arr[n//2])            # 중앙값
if len(max_nums) == 1:
    print(max_nums[0])
else:
    print(max_nums[1])
print(abs(arr[0]-arr[n-1])) # 범위